### PR TITLE
Dust shoe alarm screen

### PR DESCRIPTION
**Pull request by DennisKiselev**
# Dust shoe alarm screen

[Jira Ticket](https://yetitoolsmartsoftware.atlassian.net/browse/YTSS-2761)

## Changelog ([master doc](https://docs.google.com/document/d/1t3s6lQuUcug1Fj8E_i2gBJNvXN0-funwBArG2hsia6k/edit))
**On merging your PR, please copy the changelog to the master doc.**

## Checklist
- [x] I have completed a self review
- [ ] I have set the recent milestone
- [ ] I have tested graphical changes on all languages
- [x] I have updated the jira ticket
- [ ] I have added the relevant labels to this PR
- [ ] I have updated documentation (if applicable)
- [ ] I have run the unit tests suite and they pass

## Description

Created a dust shoe plug alarm screen that appears when the setting is enabled and the dust shoe is detected unplugged during a job or when spindle is enabled. Visually, the screen is currently just a copy of the door screen.

### Visual Test
- [ ] Not applicable
- [x] Own computer
- [ ] Console 7"
- [ ] Console 10"

### Function Test
- [ ] Not applicable
- [x] Own computer
- [ ] Console 7"
- [ ] Console 10"

### Unit Tests
- [ ] Not applicable
- [ ] Completed

### CODE REVIEW

File: src/asmcnc/skavaUI/screen_dust_shoe_alarm.py [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+)
+
+
+class DustShoeAlarmScreen(Screen):
+    poll_for_resume = None
+    return_to_screen = "home"
+    countdown_image = ObjectProperty()
+    spindle_raise_label = ObjectProperty()
+
+    def __init__(self, **kwargs):
```
Comment (archiepeterjarvis) [inline src/asmcnc/skavaUI/screen_dust_shoe_alarm.py:207]: Use positional arguments

File: src/asmcnc/skavaUI/screen_dust_shoe_alarm.py [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+        self.m = kwargs["machine"]
+        self.jd = kwargs["job"]
+        self.db = kwargs["database"]
+        self.l = kwargs["localization"]
+        self.usm = App.get_running_app().user_settings_manager
+        self.header_label.text = self.l.get_bold("Interrupt bar pushed!")
+        self.spindle_raise_label.text = (
+            self.l.get_str("Preparing to resume, please wait") + "..."
+        )
+        self.anim_spindle_label = (
```
Comment (archiepeterjarvis) [inline src/asmcnc/skavaUI/screen_dust_shoe_alarm.py:219]: Have these been tested on a pi? I experienced huge performance hits when using the Animation class.

File: tests/manual_tests/visual_screen_tests/visual_screen_tests.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 import sys, os
 
-from asmcnc.comms.logging_system.logging_system import Logger
+from src.asmcnc.comms.logging_system.logging_system import Logger
```
Comment (archiepeterjarvis) [inline tests/manual_tests/visual_screen_tests/visual_screen_tests.py:5]: This won't work when running from easycut-smartbench

File: src/asmcnc/skavaUI/screen_dust_shoe_alarm.py (hunk 2 above)
Comment (DennisKiselev) [reply src/asmcnc/skavaUI/screen_dust_shoe_alarm.py:219]: This is copied over from the door screen, and will be removed soon anyway

File: tests/manual_tests/visual_screen_tests/visual_screen_tests.py (hunk 3 above)
Comment (DennisKiselev) [reply tests/manual_tests/visual_screen_tests/visual_screen_tests.py:5]: Why not? I ran it from there in the terminal

Comment (DennisKiselev) [reply tests/manual_tests/visual_screen_tests/visual_screen_tests.py:5]: The src append happens later on

File: src/asmcnc/skavaUI/screen_dust_shoe_alarm.py (hunk 1 above)
Comment (archiepeterjarvis) [reply src/asmcnc/skavaUI/screen_dust_shoe_alarm.py:207]: Will need to be changed for Python 3, so nicer to do it now :)

File: src/asmcnc/skavaUI/screen_dust_shoe_alarm.py (hunk 2 above)
Comment (archiepeterjarvis) [reply src/asmcnc/skavaUI/screen_dust_shoe_alarm.py:219]: Cool

**Pull request merged**